tests/S3/RegionalEndpoint/ConfigurationProviderTest.php

```diff
@@ -0,0 +1,282 @@
+<?php
+
+namespace Aws\Test\S3\RegionalEndpoint;
+
+use Aws\LruArrayCache;
+use Aws\S3\RegionalEndpoint\Configuration;
+use Aws\S3\RegionalEndpoint\ConfigurationInterface;
+use Aws\S3\RegionalEndpoint\ConfigurationProvider;
+use Aws\S3\RegionalEndpoint\Exception\ConfigurationException;
+use GuzzleHttp\Promise;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \Aws\S3\RegionalEndpoint\ConfigurationProvider
+ */
+class ConfigurationProviderTest extends TestCase
+{
+    private static $originalEnv;
+
+    private $iniFile = <<<EOT
+[custom]
+s3_us_east_1_regional_endpoint = regional
```

Comment:
Should there be a case here for different casings (title, upper) of these options?  Same for ENV.

Reply:
Added commit with tests in the provider and configuration class for a non-lowercase value.